app/controllers/programs_controller.rb

```diff
@@ -6,4 +6,52 @@ def index
   def show
     @program = Program.find(params[:id])
   end
+
+  def create
+    @program = Program.new(program_params)
+
+    if @program.save
+      flash[:notice] = "Program saved"
+      redirect_to program_path(@program)
+    else
+      flash[:notice] = @program.errors.full_messages.join(". ")
+      render :new
+    end
+  end
+
+  def new
+    if current_admin.nil?
+      redirect_to root_path
+      flash[:notice] = "This portion of the site is for admins only!"
```

Comment:
Now that you're using this pattern in both `new` and `edit`, you may want to look into using a `before_action` filter to abstract away the admin authentication piece.